tests/test_glift1.cpp

```diff
@@ -0,0 +1,174 @@
+#include <boost/test/floating_point_comparison.hpp>
+#else
+#include <boost/test/tools/floating_point_comparison.hpp>
+#endif
+
+namespace Opm::Properties {
+    namespace TTag {
+        struct TestGliftTypeTag {
+            using InheritsFrom = std::tuple<EbosTypeTag>;
+        };
+    }
+}
+
+template <class TypeTag>
+std::unique_ptr<Opm::GetPropType<TypeTag, Opm::Properties::Simulator>>
+initSimulator(const char *filename)
+{
+    using Simulator = Opm::GetPropType<TypeTag, Opm::Properties::Simulator>;
+
+    std::string filename_arg = "--ecl-deck-file-name=";
+    filename_arg += filename;
+
+    const char* argv[] = {
+        "test_equil",
+        filename_arg.c_str()
+    };
+
+    Opm::setupParameters_<TypeTag>(/*argc=*/sizeof(argv)/sizeof(argv[0]), argv, /*registerParams=*/true);
+
+    return std::unique_ptr<Simulator>(new Simulator);
+}
+
+
+namespace {
+
+struct GliftFixture {
+    GliftFixture() {
+    int argc = boost::unit_test::framework::master_test_suite().argc;
+    char** argv = boost::unit_test::framework::master_test_suite().argv;
+#if HAVE_DUNE_FEM
+    Dune::Fem::MPIManager::initialize(argc, argv);
+#else
+    Dune::MPIHelper::instance(argc, argv);
+#endif
+        using TypeTag = Opm::Properties::TTag::EclFlowProblem;
+        Opm::registerAllParameters_<TypeTag>();
+    }
+};
+
+}
+
+BOOST_GLOBAL_FIXTURE(GliftFixture);
+
+BOOST_AUTO_TEST_CASE(G1)
+{
+    //using TypeTag = Opm::Properties::TTag::EclFlowProblem;
+    using TypeTag = Opm::Properties::TTag::TestGliftTypeTag;
+    //using EclProblem = Opm::EclProblem<TypeTag>;
+    //using EclWellModel = typename EclProblem::EclWellModel;
+    using WellModel = Opm::BlackoilWellModel<TypeTag>;
+    using WellState = Opm::WellStateFullyImplicitBlackoil;
+    using StdWell = Opm::StandardWell<TypeTag>;
+    using GasLiftSingleWell = Opm::GasLiftSingleWell<TypeTag>;
+    const std::string filename = "GLIFT1.DATA";
+
+    auto simulator = initSimulator<TypeTag>(filename.data());
+
+    simulator->model().applyInitialSolution();
+    simulator->setEpisodeIndex(-1);
+    simulator->setEpisodeLength(0.0);
+    simulator->startNextEpisode(/*episodeStartTime=*/0.0, /*episodeLength=*/1e30);
+    simulator->setTimeStepSize(43200);  // 12 hours
+    simulator->model().newtonMethod().setIterationIndex(0);
+    WellModel& well_model = simulator->problem().wellModel();
+    int report_step_idx = 0;
+    well_model.beginReportStep(report_step_idx);
+    well_model.beginTimeStep();
+    well_model.updatePerforationIntensiveQuantities();
+    Opm::DeferredLogger deferred_logger;
+    well_model.calculateExplicitQuantities(deferred_logger);
+    well_model.prepareTimeStep(deferred_logger);
+    well_model.updateWellControls(deferred_logger, /* check group controls */ true);
+    well_model.initPrimaryVariablesEvaluation();
+    Opm::WellInterface<TypeTag> *well_ptr = well_model.getWell("B-1H").get();
+    StdWell *std_well = dynamic_cast<StdWell *>(well_ptr);
```

Comment:
Very very good that you have made this first effort - hopefully this can be a first step towards more testable code in opm-simulators!
![image](https://user-images.githubusercontent.com/1321665/112596154-b822b980-8e0b-11eb-85eb-359f7ff73e5e.png)